src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java

```diff
@@ -42,15 +42,29 @@
 
 public abstract class DbDialect {
 
-  protected static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
-  protected static final SimpleDateFormat TIME_FORMAT = new SimpleDateFormat("HH:mm:ss.SSS");
-  protected static final SimpleDateFormat TIMESTAMP_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+  protected static final ThreadLocal<SimpleDateFormat> DATE_FORMAT = new ThreadLocal<SimpleDateFormat>() {
+    protected SimpleDateFormat initialValue() {
+      return new SimpleDateFormat("yyyy-MM-dd");
+    }
+  };
+
+  protected static final ThreadLocal<SimpleDateFormat> TIME_FORMAT = new ThreadLocal<SimpleDateFormat>() {
+    protected SimpleDateFormat initialValue() {
+      return new SimpleDateFormat("HH:mm:ss.SSS");
+    }
+  };
+
+  protected static final ThreadLocal<SimpleDateFormat> TIMESTAMP_FORMAT = new ThreadLocal<SimpleDateFormat>() {
+    protected SimpleDateFormat initialValue() {
+      return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+    }
+  };
 
   static {
     final TimeZone utc = TimeZone.getTimeZone("UTC");
-    DATE_FORMAT.setTimeZone(utc);
-    TIME_FORMAT.setTimeZone(utc);
-    TIMESTAMP_FORMAT.setTimeZone(utc);
+    DATE_FORMAT.get().setTimeZone(utc);
```

Comment:
Since this will only run on the thread where static initialization happens, don't we instead need to move these `setTimeZone` calls up into the `ThreadLocal.initialValue` calls?